transactron/_utils.py

```diff
@@ -122,24 +125,35 @@ def _graph_ccs(gr: ROGraph[T]) -> list[GraphCC[T]]:
 MethodLayout: TypeAlias = LayoutLike
 
 
-def method_def_helper(method, func: Callable[..., T], arg=None, /, **kwargs) -> T:
+def has_first_param(func: Callable[..., T], name: str, tp: type[U]) -> TypeGuard[Callable[Concatenate[U, ...], T]]:
+    parameters = signature(func).parameters
+    return (
+        len(parameters) > 1
+        and next(iter(parameters.keys())) == name
```

Comment:
Would work even shorter:
```suggestion
        and next(iter(parameters)) == name
```
but might be less explicit.